plugins/network-elements/nuage-vsp/src/com/cloud/network/resource/NuageVspResource.java

```diff
@@ -464,4 +517,10 @@ protected void isNuageVspSyncLoaded() throws Exception {
             throw new Exception(NUAGE_VSP_PLUGIN_ERROR_MESSAGE);
         }
     }
+
+    protected void isNuageVspManagerLoaded() throws Exception {
+        if (!_isNuageVspClientLoaded || _nuageVspManagerClient == null) {
+            throw new Exception(NUAGE_VSP_PLUGIN_ERROR_MESSAGE);
```

Comment:
Generally, ``Exception`` should not be thrown. Thrown the specific type of exception that causes this condition (e.g. ``IllegalStateException``).